challenges/platform/devops.md

```diff
@@ -0,0 +1,109 @@
+  17: E10FA20A:E4F4 2A5BBD5B:01BB 01 00000000:00000000 00:00000000 00000000     0        0 28072 1 0000000000000000 21 4 0 10 -1
+  18: E10FA20A:0016 3BD7E1CC:D94B 01 00000000:00000000 02:000AF7F9 00000000     0        0 27208 3 0000000000000000 22 4 29 10 -1
+  19: E10FA20A:E6A6 B358BD5B:01BB 01 00000000:00000000 00:00000000 00000000     0        0 26165 1 0000000000000000 28 4 0 10 -1
+  20: E10FA20A:D8B2 9858BD5B:0050 06 00000000:00000000 03:00001219 00000000     0        0 0 3 0000000000000000
+  ```
+
+The second and third columns contain IP addresses and port pairs in hex format. Be careful since the IP address is printed in little endian, and the port is printed in big endian (on x86 architectures).
+
+0664A8C0:1FFE translates to 10.162.15.225:55474
+
+## Level 1:
+
+### Description
+Write a program that reads `/proc/net/tcp` every 10 seconds, and reports any new connections. 
+
+Sample Output:
+```
+2021-04-28 15:28:05: New connection: 192.0.2.56:5973 -> 10.0.0.5:80
+2021-04-28 15:28:05: New connection: 203.0.113.105:31313 -> 10.0.0.5:80
+2021-04-28 15:28:15: New connection: 203.0.113.94:9208 -> 10.0.0.5:80
+2021-04-28 15:28:15: New connection: 198.51.100.245:14201 -> 10.0.0.5:80
+```
+
+Include a readme with the program that explains any dependencies and how to build and execute the program. The interview panel will build and test the program.
+
+### Questions
+1. How would you prove the code is correct?
+2. How would you make this solution better?
+3. Is it possible for this program to miss a connection?
+4. If you weren't following these requirements, how would you solve the problem of logging every new connection? 
+
+## Level 2:
+Implement all of the level 1 requirements plus:
+1. Add a Makefile or your preferred build scripting to build and test the solution.
+2. Add some tests, test for corner cases or unexpected behaviour.
+
+### Questions
+1. Why did you choose `x` to write the build automation?
+2. Is there anything else you would test if you had more time?
+3. What is the most important tool, script, or technique you have for solving problems in production? Explain why this tool/script/technique is the most important.
+4. What is the hardest technical problem or outage you've had to solve in your career? Explain what made it so difficult?
+
+## Level 3:
+Implement all of the level 1 requirements plus:
+1. Add a prometheus endpoint to report metrics on the following:
+    1. Counter - number of new connections
+
+    Tip: There are client libraries available in many languages for providing prometheus metrics, see https://prometheus.io/docs/instrumenting/clientlibs/
+2. Build the project into a docker container, provide instructions on how to execute as a container but reporting connections on the host.
+
+### Questions
+1. If you had to deploy this program to hundreds of servers, what would be your preferred method? Why?
+2. Describe a problem and how you solved that problem that you are particularly proud of?
```

Comment:
I would consider moving this to Level 2 and moving up the following to Level 3
```4. What is the hardest technical problem or outage you've had to solve in your career? Explain what made it so difficult?```

Reply:
Or cut one of these two.  I'm not sure we get that much extra signal by having both?